datastream-server/src/main/java/com/linkedin/datastream/server/DatastreamTaskImpl.java

```diff
@@ -99,6 +102,7 @@ public DatastreamTaskImpl() {
     _partitions = new ArrayList<>();
     _partitionsV2 = new ArrayList<>();
     _dependencies = new ArrayList<>();
+    computeHash();
```

Comment:
While I understand your concern, it's fairly daunting and error-prone to have to call `computeHash()` in (almost) every single method that could mutate this object (but not all of them). Precomputing hashcode is mostly useful when the type is immutable. I think it'd be more straightforward to just compute the hashcode on demand, especially that most — if not all of — Brooklin's code is not CPU bound anyways.

Reply:
+1

Reply:
Done.